x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/components/multi_text_input.tsx

```diff
@@ -0,0 +1,158 @@
+  index: number;
+  value: string;
+  onChange: (index: number, value: string) => void;
+  onDelete: (index: number) => void;
+  autoFocus?: boolean;
+  isDisabled?: boolean;
+  multi?: boolean;
+}
+
+const Row: FunctionComponent<RowProps> = React.memo(
+  ({ index, value, onChange, onDelete, autoFocus, isDisabled, multi }) => {
+    const onDeleteHandler = useCallback(() => {
+      onDelete(index);
+    }, [onDelete, index]);
+
+    const onChangeHandler = useCallback(
+      (e: ChangeEvent<HTMLInputElement>) => {
+        onChange(index, e.target.value);
+      },
+      [onChange, index]
+    );
+
+    return (
+      <EuiFlexGroup alignItems="center" gutterSize="none" responsive={false}>
+        <EuiFlexItem>
+          <EuiFieldText
+            fullWidth
+            value={value}
+            onChange={onChangeHandler}
+            autoFocus={autoFocus}
+            disabled={isDisabled}
+          />
+        </EuiFlexItem>
+        {multi && (
+          <EuiFlexItem grow={false}>
+            <EuiButtonIcon
+              color="text"
+              onClick={onDeleteHandler}
+              iconType="cross"
+              disabled={isDisabled}
+              aria-label={i18n.translate('xpack.fleet.multiTextInput.deleteRowButton', {
+                defaultMessage: 'Delete row',
+              })}
+            />
+          </EuiFlexItem>
+        )}
+      </EuiFlexGroup>
+    );
+  }
+);
+
+function defaultValue(value: string[]) {
+  return value.length > 0 ? value : [''];
+}
+
+export const MultiTextInput: FunctionComponent<Props> = ({
+  value,
+  onChange,
+  isInvalid,
+  isDisabled,
+  errors,
+}) => {
+  const [autoFocus, setAutoFocus] = useState(false);
+  const [rows, setRows] = useState(() => defaultValue(value));
+  const [previousRows, setPreviousRows] = useState(rows);
+
+  useEffect(() => {
+    if (previousRows === rows) {
+      return;
+    }
+    setPreviousRows(rows);
+    if (rows[rows.length - 1] === '') {
+      onChange(rows.slice(0, rows.length - 1));
+    } else {
+      onChange(rows);
+    }
+  }, [onChange, previousRows, rows]);
+
+  const onDeleteHandler = useCallback(
+    (idx: number) => {
+      setRows([...rows.slice(0, idx), ...rows.slice(idx + 1)]);
+    },
+    [rows]
+  );
+
+  const onChangeHandler = useCallback(
+    (idx: number, newValue: string) => {
+      const newRows = [...rows];
+      newRows[idx] = newValue;
+      setRows(newRows);
+    },
+    [rows]
+  );
+
+  const addRowHandler = useCallback(() => {
+    setAutoFocus(true);
+    setRows([...rows, '']);
+  }, [rows]);
+
+  return (
+    <>
+      <EuiFlexGroup gutterSize="s" direction="column">
+        {rows.map((row, idx) => (
+          <EuiFlexItem key={idx}>
+            <Row
+              index={idx}
+              onChange={onChangeHandler}
+              onDelete={onDeleteHandler}
+              value={row}
+              autoFocus={autoFocus}
+              isDisabled={isDisabled}
+              multi={rows.length > 1}
```

Comment:
nit: `multi` as the prop name is confusing as it is the same as the field on the var definition, can we change to `hasMultiple`, `totalRows`, or similar?